lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2912,11 +2931,20 @@ constructSharedTransaction
                 let refunds = case optionalDelegationAction of
                         Just Quit -> [W.stakeKeyDeposit pp]
                         _ -> []
+                delCerts <- case optionalDelegationAction of
+                    Nothing -> pure Nothing
+                    Just action -> do
+                        res <- liftHandler $ W.readSharedRewardAccount @n db wid
+                        when (isNothing res)$
+                            liftHandler $ throwE ErrConstructTxDelegationInvalid
+                        let path = snd $ fromJust res
```

Comment:
Same as other comment — `fromJust` shouldn't be needed.

Reply:
done